### PR TITLE
Feature/35996 - Update swatches element

### LINKED ISSUE #168: Update swatches element
Swatches looks like not styles for alpaca theme, have we design for it?

- do we need labels for color swatches?
- size swatches to update with design

### PR DESCRIPTION
**Pull request by kjugi**
+ Delete color variant and add attributes
+ Add preview to attributes variant with images
+ Simplify styles and variables

Fixes #168 
https://dev.snowdog.pro/issues/35996

### CODE REVIEW

File: components/02-elements/swatch/_swatch.scss [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
+$swatch__option-margin                     : 0 $spacer--medium $spacer--medium 0 !default;
+$swatch__option-background                 : $white !default;
+$swatch__option-color                      : $gray-dark !default;
+$swatch__option-border                     : 1px solid $white !default;
+$swatch__option-border-color-hover         : $color-primary !default;
+$swatch__option-border-color--active       : $color-primary !default;
+$swatch__option-width-screen-m--attributes : 42px !default;
+$swatch__option-height-screen-m--attributes: 48px !default;
+$swatch__option-width-screen-l--attributes : 54px !default;
+$swatch__option-height-screen-l--attributes: 64px !default;
```
Comment (anqaka) [inline components/02-elements/swatch/_swatch.scss]: in those variable we have 2 modifiers, first it's media query and we should use `small`, `medium`, `large` etc, IMO it should be like:  `$swatch__option-width--medium--attributes` etc

File: components/02-elements/swatch/_swatch.scss [hunk 2]
```diff
@@ -38,47 +29,41 @@ $swatch__border-color--color--active  : $white !default;
...
+        min-height: $swatch__option-sizing;
+        margin: $swatch__option-margin;
+        border: $swatch__option-border;
+        background: $swatch__option-background;
+        color: $swatch__option-color;
         transition: $swatch__transition;
-        color: $swatch__label-color;
         cursor: pointer;
 
         &:hover {
```
Comment (anqaka) [inline components/02-elements/swatch/_swatch.scss]: use `&:focus` with hover

File: components/02-elements/swatch/_swatch.scss [hunk 3]
```diff
@@ -38,47 +29,41 @@ $swatch__border-color--color--active  : $white !default;
...
-            border-color: $swatch__border-color--color;
-            outline: $swatch__outline--color;
+        &--active {
+            border-color: $swatch__option-border-color--active;
+        }
 
-            &:hover {
-                border-color: $swatch__border-color-hover--color;
-                outline-color: $swatch__outline-color-hover--color;
+        &--attributes {
```
Comment (anqaka) [inline components/02-elements/swatch/_swatch.scss]: so how it is finally - we should add modifiers in BEM on the beginning of block or on the end? ;)

File: components/02-elements/swatch/swatch.config.js [hunk 4]
```diff
@@ -34,30 +35,26 @@ module.exports = {
...
           },
           {
-            class: 'swatch__option--color',
-            attributes: 'style="background: #0275d8"',
-            text: ''
+            class: 'swatch__option--attributes',
+            attributes: 'style="background: url(../../images/swatch/swatch-white.jpg); background-size: cover; background-position: center center;"'
           }
         ]
       }
```
Comment (anqaka) [inline components/02-elements/swatch/swatch.config.js]: In `components/02-elements/swatch/swatch.hbs` line 2, the `h5` is used... why? The heading shouldn't be relative to the design? maybe we should move it to config

Comment (anqaka) [reply components/02-elements/swatch/swatch.config.js]: please add variant for category view

File: components/02-elements/swatch/_swatch.scss (hunk 2 above)
Comment (kjugi) [reply components/02-elements/swatch/_swatch.scss]: right

File: components/02-elements/swatch/_swatch.scss (hunk 3 above)
Comment (kjugi) [reply components/02-elements/swatch/_swatch.scss]: but it's fine. It's modifier in `swatch__option` class and it's after all mq, and hover/focus statements, then I have modifiers and in the end I have other classes (in this case there is no other classes)

File: components/02-elements/swatch/_swatch.scss (hunk 1 above)
Comment (kjugi) [reply components/02-elements/swatch/_swatch.scss]: ok

File: components/02-elements/swatch/swatch.config.js (hunk 4 above)
Comment (kjugi) [reply components/02-elements/swatch/swatch.config.js]: ok, `h5` removed. And as we discussed on chat we don't need variant for category view b/c it's the same

File: components/02-elements/swatch/_swatch.scss (hunk 3 above)
Comment (anqaka) [reply components/02-elements/swatch/_swatch.scss]: ok, my mistake, it's modifier from element 👯

File: components/02-elements/swatch/_swatch.scss [hunk 5]
```diff
@@ -38,47 +57,42 @@ $swatch__border-color--color--active  : $white !default;
...
-            background: $swatch__background--active;
-            border-color: $swatch__border-color--active;
-            color: $swatch__label-color--active;
-
-            &:hover {
-                color: $swatch__label-color--active;
-            }
+        @include mq($screen-m) {
+            min-width: $swatch__option-sizing--medium;
+            min-height: $swatch__option-sizing--medium;
```
Comment (anqaka) [inline components/02-elements/swatch/_swatch.scss]: this value is the same as `$swatch__option-sizing` use above, so we can remove it. Btw, please confirm if we shouldn't leave the swatch as a square (48x48) for that medium screen, IMO those 42px vs 48px ot's not the case. Confirm with Vasyl please

File: components/02-elements/swatch/swatch.hbs [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <div class="swatch {{ class }}" {{{ attributes }}}>
-    <h5 class="swatch__title {{ titleClass }}">
-        {{ title }}
-    </h5>
+    <{{{ heading.tag}}} class="swatch__title {{ heading.class }}">
+        {{ heading.title }}
+    </{{{ heading.tag}}}>
```
Comment (anqaka) [inline components/02-elements/swatch/swatch.hbs]: missing space after `heading.tag`, both cases

File: components/02-elements/swatch/swatch.config.js (hunk 4 above)
Comment (anqaka) [reply components/02-elements/swatch/swatch.config.js]: 👍

File: components/02-elements/swatch/swatch.hbs (hunk 6 above)
Comment (kjugi) [reply components/02-elements/swatch/swatch.hbs]: thanks

File: components/02-elements/swatch/_swatch.scss (hunk 5 above)
Comment (kjugi) [reply components/02-elements/swatch/_swatch.scss]: size swatch should be 48x48 everywhere but colour swatch is not ended yet so I leave it.

**Pull request merged**